### PR TITLE
Modernize dj-database-url

### PR DESCRIPTION
**Pull request by dcwatson**
I think this PR addresses a number of outstanding issues, namely:

* The ability to register custom backends/schemas (and simultaneously only support "blessed" Django backends by default)
* The ability to pass arbitrary OPTIONS into `config` and `parse`
* Cleaner customization of backend-specific options
* Modernize the testing matrix
* Blacken (and isort) the codebase

As I don't want to impose any more maintenance burden on you, consider this PR also as an offer to maintain this library. My original plan was to simply fork it and release under a new name, but figured it was worth offering. The current test suite is almost entirely unchanged (and passing - https://github.com/imsweb/django-dburl/actions/runs/1373144981) - it only drops pre-2.0 Django support, and registers the previously-supported backends:

https://github.com/imsweb/django-dburl/blob/modernize/test_dj_database_url.py#L12-L23

Happy to discuss further. I have a few projects at work that use this, so I'm invested in keeping it up to date, either by maintaining this library directly or maintaining my own fork under a new name.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,40 +1,39 @@
...
-on: [push]
+name: CI
+
+on: [push, pull_request]
+
 jobs:
+  checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
```
Comment (mwarkentin) [inline .github/workflows/test.yml:9]: There's a [`checkout@v2`](https://github.com/actions/checkout/blob/main/CHANGELOG.md) of this action you might want to "check out".

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,40 +1,39 @@
...
+      - name: Set up Python
+        uses: actions/setup-python@v1
```
Comment (mwarkentin) [inline .github/workflows/test.yml:11]: Similarly there's a [`setup-python@v2`](https://github.com/actions/setup-python/releases) as well.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -1,40 +1,39 @@
...
-          - python-version: 2.7
-            django-version: 2.2
-          - python-version: 2.7
-            django-version: 3.0
-          # Python 3.5 is compatible with Django 1.11 to 2.2 but not 3.0
-          - python-version: 3.5
-            django-version: 3.0
       fail-fast: false
+      matrix:
+        django-version: [2.0, 2.1, 2.2, 3.0, 3.1, 3.2]
```
Comment (mwarkentin) [inline .github/workflows/test.yml:28]: [Django 4.0](https://docs.djangoproject.com/en/4.0/releases/4.0/) is out.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -1,40 +1,39 @@
...
+        python-version: [3.6, 3.7, 3.8, 3.9]
```
Comment (mwarkentin) [inline .github/workflows/test.yml:29]: And [python 3.10](https://www.python.org/downloads/release/python-3100/).

**Comment by dcwatson** [comment]
Thanks, we'll address these when we get the repository maintenance stuff squared away.

File: .github/workflows/test.yml (hunk 4 above)
Comment (alexanderGerbik) [reply .github/workflows/test.yml:29]: Don't forget to wrap it with quotes `"3.10"`, otherwise it is interpreted as float `3.1`.

**Comment by s-i-l-k-e** [comment]
Hi, just enquiring about the state of this PR. Keen to help out where I can because I've noticed a few more outdated items in the code that'd be good to get squared away.

**Comment by dcwatson** [comment]
Now that we have the repository moved to its new home, I'll probably start merging and triaging this weekend. If you have specific fixes in mind that don't already have a PR or issue, by all means open up a new issue/PR.

**Comment by s-i-l-k-e** [comment]
Cheers, very much appreciated. Can do.

**Comment by mattseymour** [comment]
Is is possible to break this PR down into smaller chunks to make reviewing easier. My concern is this touches a lot of things and while i get it may add improvement, it is sweeping as a change. As this package is use by so many people I am worried new bugs or issues could be introduced on something which is fundimentally stable.

**Pull request closed without merging**